### PR TITLE
Hook pre commit

### PR DESCRIPTION
**Pull request by tpasternak**
Add pre-commit git hook to ensure the code is properly formatted.

### CODE REVIEW

File: pre-commit [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+   exit 1
+fi
+
+if scalafmt --quiet --test --non-interactive $files; then
+    exit 0
+else
+    echo ""
+    echo "==========================================================================="
+    echo "Scala files not formatted with scalafmt. "
+    echo "Please run 'scalafmt $rootdir/src'q"
```
Comment (PawelLipski) [inline pre-commit:21]: Stray `q`

File: pre-commit [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/sh
+
+rootdir=$(git rev-parse --show-toplevel)
+
+files=$(git diff --name-only HEAD | grep -E "[.]scala$")
```
Comment (PawelLipski) [inline pre-commit:5]: Nit: `git diff --name-only HEAD '*.scala'` should be OK as well, pls check

File: README.md [hunk 3]
```diff
@@ -11,3 +11,10 @@ Full information about Fury is available on the [Fury website](https://fury.buil
 Fury is under active development, but a binary release is not publicly
 available yet.
 
+### Code formatting
+Before committing, please remember to run auto-formatter on the code `scalafmt src/`.
+
+You can enable a pre-commit hook that will check it for you with:
+
+    ln -s ../../pre-commit .git/hooks/pre-commit
```
Comment (PawelLipski) [inline README.md:19]: Isn't it rather `ln -s pre-commit .git/hooks/pre-commit` from README.md's perspective?

**Review by propensive — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (tpasternak) [reply README.md:19]: No, it must be done from `.git/hooks` perspective even if you run if from root dir

File: pre-commit (hunk 2 above)
Comment (tpasternak) [reply pre-commit:5]: Will be fixed in next PR

File: pre-commit (hunk 1 above)
Comment (tpasternak) [reply pre-commit:21]: Will be fixed in next PR

**Pull request merged**